src/wallet/wallet.cpp

```diff
@@ -2467,7 +2477,13 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
             }
             setPresetCoins.insert(coin);
         } else {
```

Comment:
This `else` is for an `if(txid is in wallet)` ... but what we actually want to check is `if(outpoint is in wallet)`. If you try to spend an output you don't know from a wallet transaction (e.g. by sending coins to a peer then trying to spend their output), this logic will simply fail -- it will `return false;` a few lines above when trying to compute `wtx.GetSpendSize` on the outpoint.

Reply:
I think I've resolved this.